File: binary-search/javascript/binary-search.js
function binarySearch(list, item) {
  let lowerValue = 0,
      highestValue = list.length - 1;

  while (lowerValue <= highestValue) {
    let middle = Math.floor((lowerValue + highestValue) / 2);

    if (list[middle] === item) {
      return middle;
    } else if (list[middle] < item) {
      lowerValue = middle + 1;
    } else {
      highestValue = middle - 1;
    }
  }
  
  return null;
}

const list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

console.log(binarySearch(list, 11));    // null
console.log(binarySearch(list, 4));     // 3
console.log(binarySearch(list, 8));     // 7

File: binary-search/python/binary-search.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-

def binarySearch(list, item):
  lowerValue = 0
  highestValue = len(list) - 1

  while lowerValue <= highestValue:
    middle = (lowerValue + highestValue) / 2
  
    if list[middle] == item:
      return middle
    elif list[middle] > item:
      highestValue = middle - 1
    else:
      lowerValue = middle + 1
  
  return None
  
list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

print binarySearch(list, 11)    # None
print binarySearch(list, 4)     # 3
print binarySearch(list, 8)     # 7

